### PR TITLE
async/await edits to Motivation & Proposed Solution

### PR DESCRIPTION
**Pull request by kavon**
Other than some wordsmithing, I expanded the discussion about conditionally executing an async function, since it previously didn't emphasize what the problem was.

### CODE REVIEW

File: proposals/nnnn-async-await.md [hunk 1]
```diff
@@ -197,38 +200,38 @@ func processImageData2() async -> Image {
...
...rly important because *suspensions interrupt atomicity*.  For example, if an asynchronous function is running within a given context that is protected by a serial queue, reaching a suspension point means that other code can be interleaved on that same serial queue.  A classic but somewhat hackneyed example where this atomicity matters is a modeling a bank: if a deposit is credited to one account, but the operation suspends before processing a matched withdrawal, it creates a window where those funds can be double-spent.  A more germane example for many Swift programmers is a UI thread: the suspension points are the points where the UI can be shown to the user, so programs that build part of their UI and then suspend risk presenting a flickering, partially-constructed UI.  (Note that suspension points are also called out explicitly in code using explicit callbacks: the suspension happens between the point where the outer function returns and the callback starts running.)  Requiring that all suspension points are marked allows programmers to safely assume that places without suspension points will behave atomically, as well as to more easily recognize problematic non-atomic patterns.
```
Comment (kavon) [inline proposals/nnnn-async-await.md:217]: Not sure why the diff isn't highlightling well here, but I mainly added emphasis (with *) on parts I thought were important. Plus, the last sentence was reworded.

File: proposals/nnnn-async-await.md [hunk 2]
```diff
@@ -120,30 +120,32 @@ processImageData2 { result in
...
     } else {
         decodeImage { image in
-            continuation(image)
+            swizzle(image)
         }
     }
 }
 ```
 
+This pattern inverts the natural top-down organization of the function: the code that will execute in the second half of the function must appear *before* the part that executes in the first half. In addition to restructuring the entire function, we must now think carefully about captures in the continuation closure. The problem worsens as the number of conditionally-executed async functions grows, yielding what is essentially an inverted "pyramid of doom."
```
Comment (ktoso) [inline proposals/nnnn-async-await.md:144]: That's pretty well worded 👍

**Review by ktoso — APPROVED** [review]
Looks like a good improvements to me 👍

**Review by DougGregor — APPROVED** [review]
Edits look great, thank you!

**Pull request merged**